File: seeker/snippet/autoclicker.py
#date: 2022-03-15T17:12:26Z
#url: https://api.github.com/gists/3f7df83ab8c8e503fa617049e85f43ea
#owner: https://api.github.com/users/bigjango13

from time import sleep
from pynput.mouse import Button, Controller
import keyboard

mouse = Controller()

# Change these to edit the keybindings
leftClickKeybindings = ['c', 'v'] 
rightClickKeybindings = ['x', 'c']
toggleClickKeybindings = ['z', 'x']
exitKeybindings = ['esc', '1']
delay = .005

while True:
    if keyboard.is_pressed(exitKeybindings[0]) and keyboard.is_pressed(exitKeybindings[1]):
        quit()
    else:
        while keyboard.is_pressed(leftClickKeybindings[0]) and keyboard.is_pressed(leftClickKeybindings[1]): # Spam left click
            mouse.press(Button.left)
            sleep(delay)
            mouse.release(Button.left)
            sleep(delay)
        while keyboard.is_pressed(rightClickKeybindings[0]) and keyboard.is_pressed(rightClickKeybindings[1]): # Spam right click
            mouse.press(Button.right)
            sleep(delay)
            mouse.release(Button.right)
            sleep(delay)
        while keyboard.is_pressed(toggleClickKeybindings[0]) and keyboard.is_pressed(toggleClickKeybindings[1]): # Spam right and then left click
            mouse.press(Button.right)
            sleep(delay)
            mouse.release(Button.right)
            sleep(delay)
            mouse.press(Button.left)
            sleep(delay)
            mouse.release(Button.left)
            sleep(delay)

File: seeker/snippet/AutoClicker.java
//date: 2024-01-30T17:09:26Z
//url: https://api.github.com/gists/d5b8371bba291cf67613f49a43b7c8ae
//owner: https://api.github.com/users/OrangoMango

import javafx.application.Application;
import javafx.application.Platform;
import javafx.stage.Stage;
import javafx.scene.robot.Robot;
import javafx.animation.AnimationTimer;
import javafx.scene.input.MouseButton;

public class AutoClicker extends Application {
    @Override
    public void start(Stage stage) {
        Robot robot = new Robot();

        AnimationTimer timer = new AnimationTimer() {
            @Override
            public void handle(long time) {
                robot.mouseClick(MouseButton.PRIMARY);
            }
        };
        timer.start();

        new Thread(() - > {
            try {
                Thread.sleep(6000);
                timer.stop();
            } catch (InterruptedException ex) {
                ex.printStackTrace();
            }
        }).start();
    }

    public static void main(String[] args) {
        launch(args);
    }
}